coffeescript-loader/loader.js

```diff
@@ -25,41 +34,50 @@ export function resolve(specifier, context, defaultResolve) {
   return defaultResolve(specifier, context, defaultResolve);
 }
 
-export function getFormat(url, context, defaultGetFormat) {
+export async function load(url, context, defaultLoad) {
   // Now that we patched resolve to let CoffeeScript URLs through, we need to
   // tell Node.js what format such URLs should be interpreted as. Because
   // CoffeeScript transpiles into JavaScript, it should be one of the two
   // JavaScript formats: 'commonjs' or 'module'.
   if (extensionsRegex.test(url)) {
-    // CoffeeScript files can be either CommonJS or ES modules, so we want any
-    // CoffeeScript file to be treated by Node.js the same as a .js file would
-    // be at the same location (based on the "type" field in the nearest
-    // parent package.json file). To determine how Node.js would interpret an
-    // arbitrary .js file, append .js to our CoffeeScript URL and see what
-    // format Node.js returns for such a URL. If 'commonjs' is returned, a
-    // handler will need to be registered with require.extensions to process
-    // that file via the CommonJS loader.
-    const { format } = defaultGetFormat(`${url}.js`);
-    return {format};
-  }
+    // CoffeeScript files can be either CommonJS or ES modules, but since load
+    // handles both format and source based on the same url, it cannot be used
+    // for non-js files. Instead, a quick search up the filesystem for a
+    // package.json with a `type` field settles the format issue.
```

Comment:
This rewritten comment only makes sense to someone who knows the history that there used to be a separate `getFormat` hook. We could phrase it more similarly to the original:

```suggestion
    // CoffeeScript files can be either CommonJS or ES modules, so we want any
    // CoffeeScript file to be treated by Node.js the same as a .js file would
    // be at the same location. To determine how Node.js would interpret an
    // arbitrary .js file, search up the file system for the nearest parent
    // package.json file and read its "type" field.
```